### PR TITLE
remove unwanted phantom data field from image

### PR DESCRIPTION
**Pull request by kindlychung**

### CODE REVIEW

File: druid/src/widget/image.rs [hunk 1]
```diff
@@ -83,10 +80,13 @@ impl<T: Data> Widget<T> for Image<T> {
         }
     }
 
-    fn paint(&mut self, paint_ctx: &mut PaintCtx, _data: &T, _env: &Env) {
+    fn paint(&mut self, paint_ctx: &mut PaintCtx, _data: &u32, _env: &Env) {
+        // dbg!(paint_ctx.size());
+        // dbg!(self.image_data.get_size());
```
Comment (fishrockz) [inline druid/src/widget/image.rs]: Lets not merge these debug lines?

Comment (kindlychung) [reply druid/src/widget/image.rs]: sure.

File: druid/src/widget/image.rs [hunk 2]
```diff
@@ -60,18 +57,18 @@ impl<T: Data> Image<T> {
     }
 }
 
-impl<T: Data> Widget<T> for Image<T> {
-    fn event(&mut self, _ctx: &mut EventCtx, _event: &Event, _data: &mut T, _env: &Env) {}
+impl Widget<u32> for Image {
```
Comment (cmyr) [inline druid/src/widget/image.rs]: here we do need a `T`.

The rationale is that the image might need to go in a container, which is part of a widget graph that has some data type.

So it's fine to remove `T` from `Image`, but we still need `impl<T: Data> Widget<T> for Image`.

File: druid/src/widget/image.rs [hunk 3]
```diff
@@ -83,10 +80,13 @@ impl<T: Data> Widget<T> for Image<T> {
...
     }
 
-    fn paint(&mut self, paint_ctx: &mut PaintCtx, _data: &T, _env: &Env) {
+    fn paint(&mut self, paint_ctx: &mut PaintCtx, _data: &u32, _env: &Env) {
+        // dbg!(paint_ctx.size());
+        // dbg!(self.image_data.get_size());
         let offset_matrix = self
             .fill
             .affine_to_fill(paint_ctx.size(), self.image_data.get_size());
+        dbg!(offset_matrix);
```
Comment (cmyr) [inline druid/src/widget/image.rs]: ditto

**Comment by kindlychung** [comment]
> marking this as needing changes so i can better keep track of PRs; please just request a new review when it's updated. :)

Changes commited. Please have a look. @cmyr

**Pull request merged**